pkg/client/restclient/http_client.go

```diff
@@ -0,0 +1,22 @@
+// Copyright 2020 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restclient
+
+import "net/http"
+
+// HTTPClient interface
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
```

Comment:
I believe we don't need this. Just interact with the `http` package directly. It's designed to be this way.

Reply:
`HTTPClient` interface helps me to mock http call. If we directly initialize `http.Client` in calling method then we will not be able to mock its call to run testcase. This is a same reason I have to put `HTTPCli` inside `client.go`

Reply:
You can use the `httptest` package directly. I understand that the Interface approach is used out there, so that might be ok for now. This will be refactored soon.

Now, about the attribute in the client package I'd rather to remove it from there. That package is meant to be a k8s API client handler only.

Reply:
done